core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/TableSink.java

```diff
@@ -80,6 +88,28 @@ protected boolean shouldSkipCreateAtConfigure() {
       tableSinkConfig.containsMacro(Properties.Table.PROPERTY_SCHEMA_ROW_FIELD);
   }
 
+  @Override
+  public void prepareRun(BatchSinkContext context) throws DatasetManagementException {
+    super.prepareRun(context);
+    String schemaString = tableSinkConfig.getSchemaStr();
+    if (schemaString != null) {
+      try {
+        Schema schema = Schema.parseJson(schemaString);
+        if (schema.getFields() != null) {
+          FieldOperation operation =
+            new FieldWriteOperation("Write", "Writing to CDAP Table",
+                                    EndPoint.of(context.getNamespace(), tableSinkConfig.getName()),
+                                    schema.getFields().stream().map(Schema.Field::getName)
+                                      .collect(Collectors.toList()));
+          context.record(Collections.singletonList(operation));
+        }
+      } catch (IOException e) {
+        throw new IllegalStateException("Failed to parse schema", e);
```

Comment:
add a period at the end to be consistent with others.

Reply:
fixed.